### ISSUE TITLE
normalized vector

### ISSUE THREAD

**Issue by ReySadeghi** [UKPLab/sentence-transformers#712]
Hi,
is the embedding a normalized vector ?
if we normalized the embedding(vector) to norm 1, does it reduce the accuracy of finding similarity when we do cosine similarity?

**Comment by nreimers** [comment]
Vectors are usually not normalize. Normalization has no impact on cosine similarity

**Comment by timpal0l** [comment]
Is it correct to assume that plain dot product on sentence embeddings can be used instead of cosine similarity (which is just the dot product of the L2 normed vectors). The normalization does not change the order. Why do people use cosine similarity than just dot product, it is one less operation so should be faster...

**Comment by nreimers** [comment]
@timpal0l 
If the vectors are normalized (i.e. have length 1), then cosine similarity and dot product are the same.

If the vectors are not normalized, i.e. can be longer or shorter than length 1, then dot-product and cosine similarity can yield completely different rankings. In my experiences, dot product with not normalized vectors does not work well

**Comment by vikigenius** [comment]
@nreimers 

The usage pages of the `mpnet-cos` and `mpnet-dot` variants are confusing.

https://huggingface.co/sentence-transformers/multi-qa-mpnet-base-cos-v1

Says it produces normalized vectors. But the code example show us manually normalizing them like this

```python
def encode(texts):
    # Tokenize sentences
    encoded_input = tokenizer(texts, padding=True, truncation=True, return_tensors='pt')

    # Compute token embeddings
    with torch.no_grad():
        model_output = model(**encoded_input, return_dict=True)

    # Perform pooling
    embeddings = mean_pooling(model_output, encoded_input['attention_mask'])

    # Normalize embeddings
    embeddings = F.normalize(embeddings, p=2, dim=1)
    
    return embeddings

```

Obviously now we can use either dot or cos since we have manually normalized the embeddings and both will give the same result since norm is 1.

But on the page for the `mpnet-dot` model 
https://huggingface.co/sentence-transformers/multi-qa-mpnet-base-dot-v1

```python
#Encode text
def encode(texts):
    # Tokenize sentences
    encoded_input = tokenizer(texts, padding=True, truncation=True, return_tensors='pt')

    # Compute token embeddings
    with torch.no_grad():
        model_output = model(**encoded_input, return_dict=True)

    # Perform pooling
    embeddings = cls_pooling(model_output)

    return embeddings
```

We are not normalizing the embeddings and the usage suggests to use dot product on unnormalized vectors.

`scores = torch.mm(query_emb, doc_emb.transpose(0, 1))[0].cpu().tolist()` 

Isn't this contrary to 

> In my experiences, dot product with not normalized vectors does not work well

Am I misunderstanding how these models are supposed to be used ? Why can't we not use cosine as the scoring function on the mpnet-dot model ?
